components/datetime/src/pattern/mod.rs

```diff
@@ -281,3 +286,24 @@ impl Serialize for Pattern {
         }
     }
 }
+
+/// Used to represent either H11/H12, or H23/H24. Skeletons only store these
+/// hour cycles as H12 or H23.
+#[derive(Debug, PartialEq, Clone, Copy)]
+#[cfg_attr(
+    feature = "provider_serde",
+    derive(serde::Serialize, serde::Deserialize)
+)]
+pub enum CoarseHourCycle {
+    /// Can either be fields::Hour::H11 or fields::Hour::H12
+    H11H12,
+    /// Can either be fields::Hour::H23 or fields::Hour::H24
+    H23H24,
+}
+
+/// Default is required for serialization, arbitrarily pick one.
```

Comment:
Nit: The default is what is used when no locale data is available, and it is supposed to be as locale-agnostic as possible. I would therefore choose H23H24 since those patterns are less likely to have day periods in them.

Reply:
Works for me.